pkg/oam/util/helper_test.go

```diff
@@ -292,3 +292,41 @@ var _ = Describe("Test unstructured related helper utils", func() {
 		}
 	})
 })
+
+var _ = Describe("Test GenTraitName helper utils", func() {
+	It("Test generate trait name", func() {
+
+		mts := v1alpha2.ManualScalerTrait{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "ns",
+				Name:      "sample-manualscaler-trait",
+			},
+			Spec: v1alpha2.ManualScalerTraitSpec{
+				ReplicaCount: 3,
+			},
+		}
+
+		test := []struct {
+			name                string
+			template            *v1alpha2.ComponentTrait
+			collisionCount      *int32
+			otherCollisionCount *int32
+		}{
+			{
+				name:     "simple",
+				template: &v1alpha2.ComponentTrait{},
+			},
+		}
+		for _, test := range test {
+			name := fmt.Sprintf("%s-%s-%s", test.name, util.TraitPrefixKey, util.ComputeHash(test.template))
+			traitName := util.GenTraitName(test.name, test.template)
+			Expect(traitName).Should(Equal(name))
+
+			new := test.template.DeepCopy()
+			new.Trait = runtime.RawExtension{
+				Object: &mts,
```

Comment:
what does this test?

Reply:
The main test is to pass in the same `ComponentTrait` to get the same hash value.